docs/architecture/bitstream_format.rst

```diff
@@ -1,28 +1,135 @@
 Bitstream format
 ================
 
-.. todo::
-  Expand on rough notes
-
-* Specific byte pattern at beginning of file to allow hardware to determine
-  width of bus providing configuration data.
-* Rest of file is 32-bit big-endian words
-* All data before 32-bit synchronization word (0xAA995566) is ignored by
-  configuration state machine
-* Packetized format used to perform register reads/writes
-
-  * Three packet header types
-
-    * Type 0 packets exist only when performing zero-fill between rows
-    * Type 1 used for writes up to 4096 words
-    * Type 2 expands word count field to 27 bits by omitting register address
-    * Type 2 must always be proceeded by Type 1 which sets register address
-  * NOP packets are used for inserting required delays
-  * Most registers only accept 1 word of data
-  * Allowed register operations depends on interface used to send packets
-
-    * Writing LOUT via JTAG is treated as a bad command
-    * Single-frame FDRI writes via JTAG fail
+FPGAs are configured with a binary file called the bitstream.
+The bitstream carries the information which logical elements on the fabric should be configured and how in order to implement the target design.
+Moreover, it contains vital information on how to perform the configuration.
+The format of the bitstream is architecture specific, although the formats for devices of the same vendor often share a number of features.
+This chapter covers some details of the bitstream format for FPGA architectures which are currently supported in SymbiFlow's bitstream manipulation tools.
```

Comment:
This sentence is too generic.  This document is about Xilinx bitstreams.

Reply:
Yeah, but it's all true and is inside the Xilinx sub section so the sentence implicitly applies to Xilinx.